src/plugins/eth/web3.js

```diff
@@ -6,60 +6,132 @@ const https = require('https')
 
 function createWeb3(config) {
   debug.enabled = config.debug
-  
-  const web3 = new Web3(new Web3.providers.HttpProvider(
-    config.httpApiUrl,
-    {
-      agent: new https.Agent({
-        rejectUnauthorized: false, // Set to false if your HTTPS node endpoint uses a self-signed certificate
-      }),
-    }
-  ));
-  
+
+  const web3 = new Web3(providers[0], {
+    agent: new https.Agent({
+      rejectUnauthorized: false, // Set to false if your HTTPS node endpoint uses a self-signed certificate
+    }),
+  })
+
+  overrideFunctions(web3)
+  overrideFunctions(web3.eth)
+  web3.subscriptionProvider = subscriptionProvider
+
   return web3
 }
 
-function createWeb3Subscriptionable (config, eventBus) {
-  debug.enabled = config.debug;
+function createWeb3Subscribable(config, eventBus) {
+  debug.enabled = config.debug
 
   const options = {
     timeout: 1000 * 15, // ms
     // Enable auto reconnection
     reconnect: {
-        auto: true,
-        delay: 5000, // ms
-        maxAttempts: false,
-        onTimeout: false
-    }
-  };
+      auto: true,
+      delay: 5000, // ms
+      maxAttempts: false,
+      onTimeout: false,
+    },
+  }
 
-  const web3 = new Web3(new Web3.providers.WebsocketProvider(
-    config.wsApiUrl,
-    options,
-  ));
+  const web3 = new Web3(
+    new Web3.providers.WebsocketProvider(config.wsApiUrl, options)
+  )
 
   web3.currentProvider.on('connect', function () {
-    debug('Web3 provider connected');
-    eventBus.emit('web3-connection-status-changed', { connected: true });
-  });
+    debug('Web3 provider connected')
+    eventBus.emit('web3-connection-status-changed', { connected: true })
+  })
   web3.currentProvider.on('error', function (event) {
-    debug('Web3 provider connection error: ', event.type || event.message);
-    eventBus.emit('web3-connection-status-changed', { connected: false });
-  });
+    debug('Web3 provider connection error: ', event.type || event.message)
+    eventBus.emit('web3-connection-status-changed', { connected: false })
+  })
   web3.currentProvider.on('end', function (event) {
-    debug('Web3 provider connection ended: ', event.reason);
-    eventBus.emit('web3-connection-status-changed', { connected: false });
-  });
+    debug('Web3 provider connection ended: ', event.reason)
+    eventBus.emit('web3-connection-status-changed', { connected: false })
+  })
 
-  return web3;
+  return web3
 }
 
 function destroyWeb3(web3) {
   web3.currentProvider.disconnect()
 }
 
+const urls = [
+  process.env.HTTP_ETH_NODE_ADDRESS,
+  process.env.HTTP_ETH_NODE_ADDRESS2,
+  process.env.HTTP_ETH_NODE_ADDRESS3,
```

Comment:
to have values here from gitlab env we need to add variable names to these lists (for each ci file):
https://github.com/Lumerin-protocol/WalletDesktop/blob/dev/.github/workflows/auto-release-dev.yml#L80
https://github.com/Lumerin-protocol/WalletDesktop/blob/dev/.github/workflows/auto-release-stg.yml#L79
https://github.com/Lumerin-protocol/WalletDesktop/blob/dev/.github/workflows/auto-release-lmn.yml#L79

Reply:
Also, i would suggest to provide array of strings for urls, not hardcoded name with number postfix

Reply:
I would enable here an arbitrary number of nodes, as an array.